internal/lm/nvml.go

```diff
@@ -151,26 +155,27 @@ func newMigCapabilityLabeler(manager resource.Manager) (Labeler, error) {
 	return labels, nil
 }
 
-func newSharingLabeler(manager resource.Manager, config *spec.Config) Labeler {
+func newSharingLabeler(manager resource.Manager, config *spec.Config) (Labeler, error) {
 	if config == nil || config.Sharing.SharingStrategy() != spec.SharingStrategyMPS {
-		return Labels{
+		labels := Labels{
 			"nvidia.com/mps.capable": "false",
 		}
+		return labels, nil
 	}
 
 	capable, err := isMPSCapable(manager)
 	if err != nil {
-		klog.ErrorS(err, "MPS-capable check failed")
-		return Labels{
-			"nvidia.com/mps.capable": "false",
-		}
+		return nil, fmt.Errorf("failed to check MPS-capable: %w", err)
 	}
 
-	return Labels{
+	labels := Labels{
 		"nvidia.com/mps.capable": strconv.FormatBool(capable),
 	}
+	return labels, nil
 }
 
+var errMPSSharingNotSupported = errors.New("MPS sharing is not supported")
+
```

Comment:
Can we put this at the top of the file instead of embedded here in the middle of the file?

Reply:
Sure. I had it close to where it was used, but it makes more sense further up.